### PR TITLE
add: integration test for remote snapshotter

### PR DESCRIPTION
**Pull request by austinvazquez**
*Issue #, if available:*
#595 

*Description of changes:*
Adds integration test to launch a task in using a remote snapshotter running inside the microVM.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Signed-off-by: Austin Vazquez <macedonv@amazon.com>

### CODE REVIEW

File: runtime/drive_handler.go [hunk 1]
```diff
@@ -172,6 +172,11 @@ func (h *StubDriveHandler) Release(
 	return nil
 }
 
+func (h *StubDriveHandler) IsReserved(requestCtx context.Context, id string) bool {
+	_, ok := h.usedDrives[id]
+	return ok
+}
```
Comment (austinvazquez) [inline runtime/drive_handler.go:178]: Possible solution for #630. I would like to split this change and review separately from this PR.

**Comment by austinvazquez** [comment]
Relies on #641 being merged.

File: snapshotter/Makefile [hunk 2]
```diff
@@ -12,33 +12,72 @@
...
+		--init \
+		$(FIRECRACKER_CONTAINERD_TEST_IMAGE):$(DOCKER_IMAGE_TAG) \
+		"go test $(EXTRAGOARGS) -run '^$(subst integ-test-,,$@)$$'"
+
+logs:
+	mkdir logs
 
 clean:
 	- rm -f demux-snapshotter
 	- rm -f http-resolver
```
Comment (ginglis13) [inline snapshotter/Makefile:78]: `rm -f http-address-resolver`

File: snapshotter/Makefile [hunk 3]
```diff
@@ -12,33 +12,72 @@
...
+	- rm -f http-address-resolver
 
 distclean: clean
+	- rm -rf logs
 
-# Install is a noop here, for now.
-install:
-
-.PHONY: all install test clean distclean
+.PHONY: all demux-snapshotter http-address-resolver clean distclean install test integ-test
```
Comment (kzys) [inline snapshotter/Makefile:83]: https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

> A phony target is one that is not really the name of a file; rather it is just a name for a recipe to be executed when you make an explicit request.

So demux-snapshotter and http-address-resolver shouldn't be listed here.

**Review by kzys — CHANGES_REQUESTED** [review]

File: snapshotter/Makefile (hunk 3 above)
Comment (austinvazquez) [reply snapshotter/Makefile:83]: Oh no my `make`-foo is slipping.

**Review by kzys — APPROVED** [review]

**Pull request merged**